lib/diagnostic.js

```diff
@@ -5,7 +5,19 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
+const min = function (originalArray, callback) {
+  const workingArray = originalArray.slice();
```

Comment:
Since we're expecting both arguments to be numbers, calling ```slice()```, an array method, on a number will throw an error.